Shopify/dawn
# adds theme toggle for avatar widget to allow opt-out without code change

### PR Summary: 



Adds theme toggle for the avatar component on the site header.

### Why are these changes introduced?

Allows admins to disable avatar component without editing code

### What approach did you take?

Wrapped account-icon component with a theme setting check. This serves two purposes:
1. The server rendered avatar does not get rendered on page load and
2. If sign in with Shop is enabled, the avatar will not get populated post sign in 

### Other considerations

### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |   |   |   |   |


### Visual impact on existing themes


None, unless a merchant unchecks the default-to-on checkbox here. Then the account icon will not be swapped for an avatar.

### Testing steps/scenarios

- Setup the theme on a shop, ensure new checkbox is checked
- With Shop Pay and customer accounts enabled
- Sign in with Shop on customer accounts
- The account icon should change to an avatar
- Uncheck the new checkbox
- Sign in with Shop on customer accounts
- The account icon should **NOT** change to an avatar

### Demo links


- [Store](https://shop-identity-team.myshopify.com/?preview_theme_id=135984218266)
- [Editor](https://admin.shopify.com/store/shop-identity-team/themes/135984218266/editor)

### Checklist
